ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/tests/Proxy.kt

```diff
@@ -54,6 +69,51 @@ public suspend fun proxyHandler(socket: Socket) {
     }
 }
 
+@OptIn(ExperimentalTime::class)
+private suspend fun handleProxyTunnel(
+    statusLine: String,
+    input: ByteReadChannel,
+    output: ByteWriteChannel
+): Boolean {
+    require(statusLine.startsWith("CONNECT "))
+
+    output.writeStringUtf8("HTTP/1.1 200 Connection established\r\n\r\n")
+    output.flush()
+
+    val hostPort = statusLine.split(" ")[1]
+
+    val host = hostPort.substringBefore(":")
+    val port = hostPort.substringAfter(":", "80").toInt()
+
+    when (host) {
+        "localhost", "127.0.0.1", "::1" -> {
+            withTimeout(30.seconds) {
+                SelectorManager(Dispatchers.IO).use { selector ->
+                    aSocket(selector).tcp().connect(host, port).use { destination ->
+                        coroutineScope {
+                            launch {
+                                destination.openReadChannel().copyAndClose(output)
+                            }
+                            launch {
+                                input.copyAndClose(destination.openWriteChannel(true))
+                            }
+                        }
+                    }
+                }
+            }
+            return true
+        }
+        else -> {
```

Comment:
what's the point of having an empty `else` block?

Reply:
Nothing particular. Will remove, thanks.